dhall/src/Dhall.hs

```diff
@@ -1049,23 +1108,27 @@ pairFromMapEntry k v = Decoder extractOut expectedOut
         | Just key <- Dhall.Map.lookup "mapKey" kvs
         , Just value <- Dhall.Map.lookup "mapValue" kvs
             = liftA2 (,) (extract k key) (extract v value)
-    extractOut expr = typeError expectedOut expr
+    extractOut expr = typeError' expectedOut expr
 
-    expectedOut = Record (Dhall.Map.fromList [("mapKey", expected k), ("mapValue", expected v)])
+    expectedOut = (\k' v' -> Record (Dhall.Map.fromList [("mapKey", k'), ("mapValue", v')]))
```

Comment:
If you enable `ApplicativeDo`, you can write:

```haskell
expectedOut = do
    k' <- expected k
    v' <- expected v
    return (Record (Dhall.Map.fromList [("mapKey", k'), ("mapValue", v')]))
```

Reply:
Done.